<!doctype html> <html lang=en > <meta charset=UTF-8 > <meta name=viewport  content="width=device-width, initial-scale=1"> <link rel=stylesheet  href="/css/jtd.css"> <link rel=icon  href="/assets/favicon.ico"> <style> /* #148 wrap long header */ .franklin-content a.header-anchor, .franklin-toc li a { word-wrap: break-word; white-space: normal; } </style> <title>Bioinformatics</title> <script src="/libs/lunr/lunr.min.js"></script> <script src="/libs/lunr/lunr_index.js"></script> <script src="/libs/lunr/lunrclient.min.js"></script> <div class=page-wrap > <div class=side-bar > <div class=header > <a href="/" class=title > Julia <br> Package <br> Comparisons </a> </div> <label for=show-menu  class=show-menu >MENU</label> <input type=checkbox  id=show-menu  role=button > <div class=menu  id=side-menu > <form id=lunrSearchForm  name=lunrSearchForm  action="/search/index.html" method=get > <input class=search-input  name=q  placeholder="Enter search term" type=text > </form> <ul class=menu-list > <li class="menu-list-item "><a href="/" class="menu-list-link ">Home</a> </ul> <hr> <ul class=menu-list > <li class="menu-list-item active"><a href="/comparisons/bioinformatics/" class="menu-list-link active">Bioinformatics</a> <ul class=menu-list-child-list > </ul> <li class=menu-list-item ><a href="/comparisons/data_structures/" class=menu-list-link >Data Structures</a> <ul class=menu-list-child-list > <li class=menu-list-item ><a href="/comparisons/data_structures/dataframes" class=menu-list-link >DataFrames</a> <li class=menu-list-item ><a href="/comparisons/data_structures/graphs" class=menu-list-link >Graphs</a> </ul> <li class=menu-list-item ><a href="/comparisons/performance_enhancement/loop_acceleration/" class=menu-list-link >Performance Enhancement</a> <ul class=menu-list-child-list > <li class=menu-list-item ><a href="/comparisons/performance_enhancement/loop_acceleration" class=menu-list-link >Loop acceleration</a> <li class=menu-list-item ><a href="/comparisons/performance_enhancement/multithreading" class=menu-list-link >Multithreading</a> <li class=menu-list-item ><a href="/comparisons/performance_enhancement/types_compactification" class=menu-list-link >Types compactification</a> </ul> <li class=menu-list-item ><a href="/comparisons/math/bspline/" class=menu-list-link >Math</a> <ul class=menu-list-child-list > <li class=menu-list-item ><a href="/comparisons/math/bspline" class=menu-list-link >B-spline</a> <li class=menu-list-item ><a href="/comparisons/math/differentiation" class=menu-list-link >Differentiation</a> <li class=menu-list-item ><a href="/comparisons/math/einsum" class=menu-list-link >Einsum</a> <li class=menu-list-item ><a href="/comparisons/math/intervals" class=menu-list-link >Intervals</a> <li class=menu-list-item ><a href="/comparisons/math/linear_solvers" class=menu-list-link >Linear Solvers</a> <li class=menu-list-item ><a href="/comparisons/math/nonlinear_solvers" class=menu-list-link >Nonlinear Solvers</a> <li class=menu-list-item ><a href="/comparisons/math/quaternions" class=menu-list-link >Quaternions</a> </ul> <li class=menu-list-item ><a href="/comparisons/physics/" class=menu-list-link >Physics</a> <ul class=menu-list-child-list > <li class=menu-list-item ><a href="/comparisons/physics/physical_constants" class=menu-list-link >Physical Constants</a> <li class=menu-list-item ><a href="/comparisons/physics/units" class=menu-list-link >Units</a> </ul> <li class=menu-list-item ><a href="/comparisons/machine_learning/machine_learning/" class=menu-list-link >Machine Learning</a> <ul class=menu-list-child-list > <li class=menu-list-item ><a href="/comparisons/machine_learning/machine_learning" class=menu-list-link >Machine Learning</a> <li class=menu-list-item ><a href="/comparisons/machine_learning/machine_learning_datasets" class=menu-list-link >Machine Learning Datasets</a> <li class=menu-list-item ><a href="/comparisons/machine_learning/neural_networks" class=menu-list-link >Neural Networks</a> </ul> <li class=menu-list-item ><a href="/comparisons/geometry/triangulations_or_tessellations/" class=menu-list-link >Geometry</a> <ul class=menu-list-child-list > <li class=menu-list-item ><a href="/comparisons/geometry/triangulations_or_tessellations" class=menu-list-link >Triangulations and Tessellations</a> </ul> <li class=menu-list-item ><a href="/comparisons/graphics/plotting/" class=menu-list-link >Graphics</a> <ul class=menu-list-child-list > <li class=menu-list-item ><a href="/comparisons/graphics/plotting" class=menu-list-link >Plotting</a> </ul> <li class=menu-list-item ><a href="/comparisons/simulation/agentbasedmodelling/" class=menu-list-link >Simulation</a> <ul class=menu-list-child-list > <li class=menu-list-item ><a href="/comparisons/simulation/agentbasedmodelling" class=menu-list-link >Agent Based Modelling</a> <li class=menu-list-item ><a href="/comparisons/simulation/control_systems" class=menu-list-link >Control Systems</a> <li class=menu-list-item ><a href="/comparisons/simulation/fluid_dynamics" class=menu-list-link >Fluid dynamics</a> </ul> <li class=menu-list-item ><a href="/comparisons/dataio/" class=menu-list-link >Data I/O</a> <ul class=menu-list-child-list > <li class=menu-list-item ><a href="/comparisons/dataio/analogio" class=menu-list-link >Analog I/O</a> <li class=menu-list-item ><a href="/comparisons/dataio/fileio" class=menu-list-link >File I/O</a> <li class=menu-list-item ><a href="/comparisons/dataio/networkio" class=menu-list-link >Network I/O</a> </ul> <li class=menu-list-item ><a href="/comparisons/interoperability/" class=menu-list-link >Interoperability</a> <ul class=menu-list-child-list > <li class=menu-list-item ><a href="/comparisons/interoperability/c_cpp" class=menu-list-link >C & C++</a> <li class=menu-list-item ><a href="/comparisons/interoperability/python" class=menu-list-link >Python</a> </ul> <li class=menu-list-item ><a href="/comparisons/utility/notebooks/" class=menu-list-link >Utility</a> <ul class=menu-list-child-list > <li class=menu-list-item ><a href="/comparisons/utility/notebooks" class=menu-list-link >Notebooks</a> <li class=menu-list-item ><a href="/comparisons/utility/package_templates" class=menu-list-link >Package Templates</a> <li class=menu-list-item ><a href="/comparisons/utility/piping" class=menu-list-link >Piping</a> <li class=menu-list-item ><a href="/comparisons/utility/redefinable_structs" class=menu-list-link >Redefinable Structs</a> </ul> <li class=menu-list-item ><a href="/comparisons/web/" class=menu-list-link >Web</a> <ul class=menu-list-child-list > <li class=menu-list-item ><a href="/comparisons/web/static_websites" class=menu-list-link >Static Websites</a> <li class=menu-list-item ><a href="/comparisons/web/web_apps" class=menu-list-link >Web Apps</a> </ul> <li class=menu-list-item ><a href="/comparisons/testing/about_testing/" class=menu-list-link >Testing</a> <ul class=menu-list-child-list > <li class=menu-list-item ><a href="/comparisons/testing/about_testing" class=menu-list-link >About Testing</a> <li class=menu-list-item ><a href="/comparisons/testing/profiling" class=menu-list-link >Profiling</a> <li class=menu-list-item ><a href="/comparisons/testing/property_based_testing" class=menu-list-link >Property Based Testing</a> <li class=menu-list-item ><a href="/comparisons/testing/runtime" class=menu-list-link >Runtime</a> <li class=menu-list-item ><a href="/comparisons/testing/unit_testing" class=menu-list-link >Unit Testing</a> </ul> <li class="menu-list-item "><a href="/comparisons/metrics/" class="menu-list-link ">Metrics</a> <li class="menu-list-item "><a href="/comparisons/probabilistic_programming/" class="menu-list-link ">Probabilistic Programming</a> <li class="menu-list-item "><a href="/comparisons/signal_processing/" class="menu-list-link ">Signal Processing</a> <li class="menu-list-item "><a href="/comparisons/uncertainty_propagation/" class="menu-list-link ">Uncertainty Propagation</a> </ul> <hr> <ul class=menu-list > <li class="menu-list-item "><a href="/contributing" class="menu-list-link ">Contributing</a> </ul> </div> <div class=footer > </div> </div> <div class=main-content-wrap > <div class=main-content > <div class=main-header > <a id=github  href="https://github.com/JuliaPackageComparisons/JuliaPackageComparisons.github.io/blob/main/docs/comparisons/bioinformatics/index.md">Page source</a> <span style="width:30px; text-align: center;color:lightgray;">|</span> <a id=github  href="https://github.com/JuliaPackageComparisons/JuliaPackageComparisons.github.io">GitHub repository</a> </div> <div class=franklin-content > <h1 id=bioinformatics ><a href="#bioinformatics" class=header-anchor >Bioinformatics</a></h1> <div class=badge ><img src="https://avatars.githubusercontent.com/u/6486271?s&#61;100&amp;v&#61;4" alt=BioJulia  /></div> <p>The <a href="https://github.com/biojulia">BioJulia</a> organization collects a lot of great packages related to bioinformatics.</p> <h1 id=file_parsers ><a href="#file_parsers" class=header-anchor >File Parsers</a></h1> <p>An important task in bioinformatics is parsing files in various standard formats. Here we list some file formats and packages with parsers:</p> <table><tr><th align=right >Format<th align=right >Extensions<th align=right >Description<th align=right >Packages<tr><td align=right ><a href="https://en.wikipedia.org/wiki/FASTA_format">FASTA</a><td align=right >.fas, .fasta, .fa<td align=right >DNA or protein sequences without annotations<td align=right ><a href="https://github.com/BioJulia/FASTX.jl">FASTX</a><tr><td align=right ><a href="https://en.wikipedia.org/wiki/FASTQ_format">FASTQ</a><td align=right >.fq, .fastq<td align=right >DNA sequences with quality information<td align=right ><a href="https://github.com/BioJulia/FASTX.jl">FASTX</a><tr><td align=right ><a href="https://en.wikipedia.org/wiki/GenBank">GENBANK</a><td align=right >.gb, .gbk<td align=right >DNA or protein sequences with annotations<td align=right ><a href="https://github.com/BioJulia/GenomicAnnotations.jl">GenomicAnnotations.jl</a><tr><td align=right >EMBL<td align=right >.embl<td align=right >DNA or protein sequences with annotations<td align=right ><a href="https://github.com/BioJulia/GenomicAnnotations.jl">GenomicAnnotations.jl</a><tr><td align=right ><a href="https://en.wikipedia.org/wiki/SAM_&#40;file_format&#41;">SAM</a><td align=right >.sam<td align=right >Aligned DNA sequences &#40;typically from read mapping&#41;. Text based.<td align=right ><a href="https://github.com/BioJulia/XAM.jl">XAM.jl</a><tr><td align=right ><a href="https://en.wikipedia.org/wiki/BAM_&#40;file_format&#41;">BAM</a><td align=right >.bam<td align=right >Aligned DNA sequences &#40;typically from read mapping&#41;. Binary.<td align=right ><a href="https://github.com/BioJulia/XAM.jl">XAM.jl</a><tr><td align=right ><a href="https://en.wikipedia.org/wiki/Protein_Data_Bank_&#40;file_format&#41;">PDB</a><td align=right >.pdb<td align=right >Protein 3D structure.<td align=right ><a href="https://github.com/BioJulia/BioStructures.jl">BioStructures.jl</a>, <a href="https://github.com/diegozea/MIToS.jl">MIToS</a><tr><td align=right ><a href="https://en.wikipedia.org/wiki/Macromolecular_Crystallographic_Information_File">mmCIF</a><td align=right ><td align=right >Macromolecular Crystallographic Information File &#40;mmCIF&#41; also known as PDBx/mmCIF is a standard text file format for representing macromolecular structure data<td align=right ><a href="https://github.com/BioJulia/BioStructures.jl">BioStructures.jl</a>, <a href="https://github.com/diegozea/MIToS.jl">MIToS</a><tr><td align=right ><a href="https://github.com/rcsb/mmtf">MMTF</a><td align=right ><td align=right >MacroMolecular Transmission Format &#40;MMTF&#41; is a binary encoding of biological structures.<td align=right ><a href="https://github.com/BioJulia/BioStructures.jl">BioStructures.jl</a><tr><td align=right ><a href="https://github.com/PDB-REDO/dssp">DSSP</a><td align=right ><td align=right >Protein Secondary Structure<td align=right ><a href="https://github.com/BioJulia/ProteinSecondaryStructures.jl">ProteinSecondaryStructures.jl</a><tr><td align=right ><a href="https://webclu.bio.wzw.tum.de/stride/">STRIDE</a><td align=right ><td align=right >Protein Secondary Structure<td align=right ><a href="https://github.com/BioJulia/ProteinSecondaryStructures.jl">ProteinSecondaryStructures.jl</a><tr><td align=right ><a href="https://github.com/slimsuite/pafscaff">PAF</a><td align=right >.paf<td align=right >Pairwise mApping Format.<td align=right ><a href="https://github.com/BioJulia/PairwiseMappingFormat.jl">PairwiseMappingFormat.jl</a><tr><td align=right ><a href="https://en.wikipedia.org/wiki/Stockholm_format">Stockholm</a><td align=right >.sto, .stk, .stockholm<td align=right >Stockholm format is a multiple sequence alignment format used by Pfam, Rfam and Dfam<td align=right ><a href="https://github.com/diegozea/MIToS.jl">MIToS.jl</a><tr><td align=right ><a href="https://en.wikipedia.org/wiki/FASTA_format">A3M</a><td align=right >.fas<td align=right >A2M/A3M are a family of FASTA-derived formats used for sequence alignments<td align=right ><a href="https://github.com/diegozea/MIToS.jl">MIToS.jl</a><tr><td align=right ><a href="https://www.bioinformatics.nl/tools/crab_pir.html">PIR</a><td align=right >.pir<td align=right >Multiple sequence alignment format<td align=right ><a href="https://github.com/diegozea/MIToS.jl">MIToS.jl</a></table> <h1 id=data_structures ><a href="#data_structures" class=header-anchor >Data Structures</a></h1> <p>The basic data structures for representing DNA, RNA and protein sequences are in <a href="https://github.com/BioJulia/BioSequences.jl">BioSequences.jl</a></p> <h1 id=pairwise_sequence_alignments ><a href="#pairwise_sequence_alignments" class=header-anchor >Pairwise Sequence Alignments</a></h1> <p>A core task in bioinformatics is aligning sequences. This can be done with <a href="https://github.com/BioJulia/BioAlignments.jl">BioAlignments.jl</a> which includes algorithms for the following pairwise alignment types:</p> <ul> <li><p>GlobalAlignment: global-to-global alignment</p> <li><p>SemiGlobalAlignment: local-to-global alignment</p> <li><p>LocalAlignment: local-to-local alignment</p> <li><p>OverlapAlignment: end-free alignment</p> </ul> <h1 id=multiple_sequence_alignment_msa ><a href="#multiple_sequence_alignment_msa" class=header-anchor >Multiple Sequence Alignment &#40;MSA&#41;</a></h1> <p>I&#39;m not aware of tools in Julia to compute multiple sequence alignment, but <a href="https://github.com/diegozea/MIToS.jl">MIToS.jl</a> can read the most common MSA formats: stockholm, FASTA, A3M, A2M, PIR or Raw format</p> <h1 id=package_descriptions ><a href="#package_descriptions" class=header-anchor >Package descriptions</a></h1> <h2 id=biosequencesjl ><a href="#biosequencesjl" class=header-anchor >BioSequences.jl</a></h2> <div class=badge > <a href="https://github.com/BioJulia/BioSequences.jl"><img src="https://img.shields.io/github/stars/BioJulia/BioSequences.jl?style=social" alt="GitHub Repo stars"></a> <a href="https://juliahub.com/ui/Packages/General/BioSequences?t=2"><img src="https://juliahub.com/docs/General/BioSequences/stable/deps.svg" alt=deps ></a> <a href="https://pkgs.genieframework.com?packages=BioSequences"><img src="https://shields.io/endpoint?url=https://pkgs.genieframework.com/api/v1/badge/BioSequences" alt="BioSequences Downloads"></a> <br> <a href="https://BioJulia.github.io/BioSequences.jl/stable"><img src="https://img.shields.io/badge/docs-stable-blue.svg" alt=Stable ></a> <a href="https://BioJulia.github.io/BioSequences.jl/dev"><img src="https://img.shields.io/badge/docs-dev-blue.svg" alt=Dev ></a> <a href="https://github.com/BioJulia/BioSequences.jl/commits/master/"><img src="https://img.shields.io/github/last-commit/BioJulia/BioSequences.jl/master" alt="GitHub last commit (branch)"></a> <a href="https://juliahub.com/ui/Packages/General/BioSequences"><img src="https://juliahub.com/docs/General/BioSequences/stable/version.svg" alt=version ></a> <a href="https://codecov.io/gh/BioJulia/BioSequences.jl"><img src="https://codecov.io/gh/BioJulia/BioSequences.jl/branch/master/graph/badge.svg" alt=Coverage ></a> </div> <blockquote> <p>Biological sequences for the julia language</p> </blockquote> <p><a href="https://github.com/BioJulia/BioSequences.jl/">BioSequences.jl</a> BioSequences provides data types and methods for common operations with biological sequences, including DNA, RNA, and amino acid sequences.</p> <p>It can do sequence search and pattern matching in sequences, and compute simple sequence statistics.</p> <h2 id=bioalignmentsjl ><a href="#bioalignmentsjl" class=header-anchor >BioAlignments.jl</a></h2> <div class=badge > <a href="https://github.com/BioJulia/BioAlignments.jl"><img src="https://img.shields.io/github/stars/BioJulia/BioAlignments.jl?style=social" alt="GitHub Repo stars"></a> <a href="https://juliahub.com/ui/Packages/General/BioAlignments?t=2"><img src="https://juliahub.com/docs/General/BioAlignments/stable/deps.svg" alt=deps ></a> <a href="https://pkgs.genieframework.com?packages=BioAlignments"><img src="https://shields.io/endpoint?url=https://pkgs.genieframework.com/api/v1/badge/BioAlignments" alt="BioAlignments Downloads"></a> <br> <a href="https://BioJulia.github.io/BioAlignments.jl/stable"><img src="https://img.shields.io/badge/docs-stable-blue.svg" alt=Stable ></a> <a href="https://BioJulia.github.io/BioAlignments.jl/dev"><img src="https://img.shields.io/badge/docs-dev-blue.svg" alt=Dev ></a> <a href="https://github.com/BioJulia/BioAlignments.jl/commits/master/"><img src="https://img.shields.io/github/last-commit/BioJulia/BioAlignments.jl/master" alt="GitHub last commit (branch)"></a> <a href="https://juliahub.com/ui/Packages/General/BioAlignments"><img src="https://juliahub.com/docs/General/BioAlignments/stable/version.svg" alt=version ></a> <a href="https://codecov.io/gh/BioJulia/BioAlignments.jl"><img src="https://codecov.io/gh/BioJulia/BioAlignments.jl/branch/master/graph/badge.svg" alt=Coverage ></a> </div> <blockquote> <p>Sequence alignment tools </p> </blockquote> <p><a href="https://github.com/BioJulia/BioAlignments.jl/">BioAlignments.jl</a> provides sequence alignment algorithms and data structures. It includes algorithms for the following pairwise alignment types:</p> <ul> <li><p>GlobalAlignment: global-to-global alignment</p> <li><p>SemiGlobalAlignment: local-to-global alignment</p> <li><p>LocalAlignment: local-to-local alignment</p> <li><p>OverlapAlignment: end-free alignment</p> </ul> <h2 id=genomicfeaturesjl ><a href="#genomicfeaturesjl" class=header-anchor >GenomicFeatures.jl</a></h2> <div class=badge > <a href="https://github.com/BioJulia/GenomicFeatures.jl"><img src="https://img.shields.io/github/stars/BioJulia/GenomicFeatures.jl?style=social" alt="GitHub Repo stars"></a> <a href="https://juliahub.com/ui/Packages/General/GenomicFeatures?t=2"><img src="https://juliahub.com/docs/General/GenomicFeatures/stable/deps.svg" alt=deps ></a> <a href="https://pkgs.genieframework.com?packages=GenomicFeatures"><img src="https://shields.io/endpoint?url=https://pkgs.genieframework.com/api/v1/badge/GenomicFeatures" alt="GenomicFeatures Downloads"></a> <br> <a href="https://BioJulia.github.io/GenomicFeatures.jl/stable"><img src="https://img.shields.io/badge/docs-stable-blue.svg" alt=Stable ></a> <a href="https://BioJulia.github.io/GenomicFeatures.jl/dev"><img src="https://img.shields.io/badge/docs-dev-blue.svg" alt=Dev ></a> <a href="https://github.com/BioJulia/GenomicFeatures.jl/commits/master/"><img src="https://img.shields.io/github/last-commit/BioJulia/GenomicFeatures.jl/master" alt="GitHub last commit (branch)"></a> <a href="https://juliahub.com/ui/Packages/General/GenomicFeatures"><img src="https://juliahub.com/docs/General/GenomicFeatures/stable/version.svg" alt=version ></a> <a href="https://codecov.io/gh/BioJulia/GenomicFeatures.jl"><img src="https://codecov.io/gh/BioJulia/GenomicFeatures.jl/branch/master/graph/badge.svg" alt=Coverage ></a> </div> <blockquote> <p>Tools for genomic features in Julia.</p> </blockquote> <p><a href="https://github.com/BioJulia/GenomicFeatures.jl/">GenomicFeatures.jl</a></p> <h2 id=genomicannotationsjl ><a href="#genomicannotationsjl" class=header-anchor >GenomicAnnotations.jl</a></h2> <div class=badge > <a href="https://github.com/BioJulia/GenomicAnnotations.jl"><img src="https://img.shields.io/github/stars/BioJulia/GenomicAnnotations.jl?style=social" alt="GitHub Repo stars"></a> <a href="https://juliahub.com/ui/Packages/General/GenomicAnnotations?t=2"><img src="https://juliahub.com/docs/General/GenomicAnnotations/stable/deps.svg" alt=deps ></a> <a href="https://pkgs.genieframework.com?packages=GenomicAnnotations"><img src="https://shields.io/endpoint?url=https://pkgs.genieframework.com/api/v1/badge/GenomicAnnotations" alt="GenomicAnnotations Downloads"></a> <br> <a href="https://BioJulia.github.io/GenomicAnnotations.jl/stable"><img src="https://img.shields.io/badge/docs-stable-blue.svg" alt=Stable ></a> <a href="https://BioJulia.github.io/GenomicAnnotations.jl/dev"><img src="https://img.shields.io/badge/docs-dev-blue.svg" alt=Dev ></a> <a href="https://github.com/BioJulia/GenomicAnnotations.jl/commits/master/"><img src="https://img.shields.io/github/last-commit/BioJulia/GenomicAnnotations.jl/master" alt="GitHub last commit (branch)"></a> <a href="https://juliahub.com/ui/Packages/General/GenomicAnnotations"><img src="https://juliahub.com/docs/General/GenomicAnnotations/stable/version.svg" alt=version ></a> <a href="https://codecov.io/gh/BioJulia/GenomicAnnotations.jl"><img src="https://codecov.io/gh/BioJulia/GenomicAnnotations.jl/branch/master/graph/badge.svg" alt=Coverage ></a> </div> <blockquote> <p>GenomicAnnotations is a package for reading, modifying, and writing genomic annotations in the GenBank, GFF3, GFF2/GTF, and EMBL file formats.</p> </blockquote> <p><a href="https://github.com/BioJulia/GenomicAnnotations.jl">GenomicAnnotations.jl</a></p> <h2 id=biostructuresjl ><a href="#biostructuresjl" class=header-anchor >BioStructures.jl</a></h2> <blockquote> <p>A Julia package to read, write and manipulate macromolecular structures </p> </blockquote> <p><div class=badge > <a href="https://github.com/BioJulia/BioStructures.jl"><img src="https://img.shields.io/github/stars/BioJulia/BioStructures.jl?style=social" alt="GitHub Repo stars"></a> <a href="https://juliahub.com/ui/Packages/General/BioStructures?t=2"><img src="https://juliahub.com/docs/General/BioStructures/stable/deps.svg" alt=deps ></a> <a href="https://pkgs.genieframework.com?packages=BioStructures"><img src="https://shields.io/endpoint?url=https://pkgs.genieframework.com/api/v1/badge/BioStructures" alt="BioStructures Downloads"></a> <br> <a href="https://BioJulia.github.io/BioStructures.jl/stable"><img src="https://img.shields.io/badge/docs-stable-blue.svg" alt=Stable ></a> <a href="https://BioJulia.github.io/BioStructures.jl/dev"><img src="https://img.shields.io/badge/docs-dev-blue.svg" alt=Dev ></a> <a href="https://github.com/BioJulia/BioStructures.jl/commits/master/"><img src="https://img.shields.io/github/last-commit/BioJulia/BioStructures.jl/master" alt="GitHub last commit (branch)"></a> <a href="https://juliahub.com/ui/Packages/General/BioStructures"><img src="https://juliahub.com/docs/General/BioStructures/stable/version.svg" alt=version ></a> <a href="https://codecov.io/gh/BioJulia/BioStructures.jl"><img src="https://codecov.io/gh/BioJulia/BioStructures.jl/branch/master/graph/badge.svg" alt=Coverage ></a> </div> <a href="https://github.com/BioJulia/BioStructures.jl/">BioStructures.jl</a></p> <p>From the package README:</p> <p>BioStructures provides functionality to read, write and manipulate macromolecular structures, in particular proteins. <a href="https://www.rcsb.org/pdb/home/home.do">Protein Data Bank</a> &#40;PDB&#41;, mmCIF and MMTF format files can be read in to a hierarchical data structure. Spatial calculations and functions to access the PDB are also provided. It compares favourably in terms of performance to other PDB parsers - see some <a href="https://github.com/jgreener64/pdb-benchmarks">benchmarks online</a> - and should be lightweight enough to build other packages on top of.</p> <h2 id=geneticvariationjl ><a href="#geneticvariationjl" class=header-anchor >GeneticVariation.jl</a></h2> <div class=badge > <a href="https://github.com/BioJulia/GeneticVariation.jl"><img src="https://img.shields.io/github/stars/BioJulia/GeneticVariation.jl?style=social" alt="GitHub Repo stars"></a> <a href="https://juliahub.com/ui/Packages/General/GeneticVariation?t=2"><img src="https://juliahub.com/docs/General/GeneticVariation/stable/deps.svg" alt=deps ></a> <a href="https://pkgs.genieframework.com?packages=GeneticVariation"><img src="https://shields.io/endpoint?url=https://pkgs.genieframework.com/api/v1/badge/GeneticVariation" alt="GeneticVariation Downloads"></a> <br> <a href="https://BioJulia.github.io/GeneticVariation.jl/stable"><img src="https://img.shields.io/badge/docs-stable-blue.svg" alt=Stable ></a> <a href="https://BioJulia.github.io/GeneticVariation.jl/dev"><img src="https://img.shields.io/badge/docs-dev-blue.svg" alt=Dev ></a> <a href="https://github.com/BioJulia/GeneticVariation.jl/commits/master/"><img src="https://img.shields.io/github/last-commit/BioJulia/GeneticVariation.jl/master" alt="GitHub last commit (branch)"></a> <a href="https://juliahub.com/ui/Packages/General/GeneticVariation"><img src="https://juliahub.com/docs/General/GeneticVariation/stable/version.svg" alt=version ></a> <a href="https://codecov.io/gh/BioJulia/GeneticVariation.jl"><img src="https://codecov.io/gh/BioJulia/GeneticVariation.jl/branch/master/graph/badge.svg" alt=Coverage ></a> </div> <blockquote> <p>Datastructures and algorithms for working with genetic variation</p> </blockquote> <p><a href="https://github.com/BioJulia/GeneticVariation.jl/">GeneticVariation.jl</a></p> <p>From the package README:</p> <p>GeneticVariation provides types and methods for working with datasets of genetic variation. It provides a VCF and BCF parser, as well as methods for working with variation in sequences such as evolutionary distance computation, and counting different mutation types.</p> <h2 id=phylogeniesjl ><a href="#phylogeniesjl" class=header-anchor >Phylogenies.jl</a></h2> <div class=badge > <a href="https://github.com/BioJulia/Phylogenies.jl"><img src="https://img.shields.io/github/stars/BioJulia/Phylogenies.jl?style=social" alt="GitHub Repo stars"></a> <a href="https://juliahub.com/ui/Packages/General/Phylogenies?t=2"><img src="https://juliahub.com/docs/General/Phylogenies/stable/deps.svg" alt=deps ></a> <a href="https://pkgs.genieframework.com?packages=Phylogenies"><img src="https://shields.io/endpoint?url=https://pkgs.genieframework.com/api/v1/badge/Phylogenies" alt="Phylogenies Downloads"></a> <br> <a href="https://BioJulia.github.io/Phylogenies.jl/stable"><img src="https://img.shields.io/badge/docs-stable-blue.svg" alt=Stable ></a> <a href="https://BioJulia.github.io/Phylogenies.jl/dev"><img src="https://img.shields.io/badge/docs-dev-blue.svg" alt=Dev ></a> <a href="https://github.com/BioJulia/Phylogenies.jl/commits/master/"><img src="https://img.shields.io/github/last-commit/BioJulia/Phylogenies.jl/master" alt="GitHub last commit (branch)"></a> <a href="https://juliahub.com/ui/Packages/General/Phylogenies"><img src="https://juliahub.com/docs/General/Phylogenies/stable/version.svg" alt=version ></a> <a href="https://codecov.io/gh/BioJulia/Phylogenies.jl"><img src="https://codecov.io/gh/BioJulia/Phylogenies.jl/branch/master/graph/badge.svg" alt=Coverage ></a> </div> <blockquote> <p>The BioJulia package for working with phylogenetic trees and geneologies. </p> </blockquote> <p><a href="https://github.com/BioJulia/Phylogenies.jl/">Phylogenies.jl</a></p> <p>This looks stale.</p> <p>From the package README:</p> <p>A julia package providing an abstract type and interface for phylogenies, a concrete phylogeny type implementation, and higher-level methods for working with phylogenies.</p> <p>In development.</p> <h2 id=genomegraphsjl ><a href="#genomegraphsjl" class=header-anchor >GenomeGraphs.jl</a></h2> <div class=badge > <a href="https://github.com/BioJulia/GenomeGraphs.jl"><img src="https://img.shields.io/github/stars/BioJulia/GenomeGraphs.jl?style=social" alt="GitHub Repo stars"></a> <a href="https://juliahub.com/ui/Packages/General/GenomeGraphs?t=2"><img src="https://juliahub.com/docs/General/GenomeGraphs/stable/deps.svg" alt=deps ></a> <a href="https://pkgs.genieframework.com?packages=GenomeGraphs"><img src="https://shields.io/endpoint?url=https://pkgs.genieframework.com/api/v1/badge/GenomeGraphs" alt="GenomeGraphs Downloads"></a> <br> <a href="https://BioJulia.github.io/GenomeGraphs.jl/stable"><img src="https://img.shields.io/badge/docs-stable-blue.svg" alt=Stable ></a> <a href="https://BioJulia.github.io/GenomeGraphs.jl/dev"><img src="https://img.shields.io/badge/docs-dev-blue.svg" alt=Dev ></a> <a href="https://github.com/BioJulia/GenomeGraphs.jl/commits/master/"><img src="https://img.shields.io/github/last-commit/BioJulia/GenomeGraphs.jl/master" alt="GitHub last commit (branch)"></a> <a href="https://juliahub.com/ui/Packages/General/GenomeGraphs"><img src="https://juliahub.com/docs/General/GenomeGraphs/stable/version.svg" alt=version ></a> <a href="https://codecov.io/gh/BioJulia/GenomeGraphs.jl"><img src="https://codecov.io/gh/BioJulia/GenomeGraphs.jl/branch/master/graph/badge.svg" alt=Coverage ></a> </div> <blockquote> <p>A modern genomics framework for julia</p> </blockquote> <p><a href="https://github.com/BioJulia/GenomeGraphs.jl">GenomeGraphs.jl</a></p> <p>From the package README:</p> <p>GenomeGraphs provides a representation of sequence graphs. Such graphs represent genome assemblies and population graphs of genotypes/haplotypes and variation.</p> <h2 id=bioservicesjl ><a href="#bioservicesjl" class=header-anchor >BioServices.jl</a></h2> <div class=badge > <a href="https://github.com/BioJulia/BioServices.jl"><img src="https://img.shields.io/github/stars/BioJulia/BioServices.jl?style=social" alt="GitHub Repo stars"></a> <a href="https://juliahub.com/ui/Packages/General/BioServices?t=2"><img src="https://juliahub.com/docs/General/BioServices/stable/deps.svg" alt=deps ></a> <a href="https://pkgs.genieframework.com?packages=BioServices"><img src="https://shields.io/endpoint?url=https://pkgs.genieframework.com/api/v1/badge/BioServices" alt="BioServices Downloads"></a> <br> <a href="https://BioJulia.github.io/BioServices.jl/stable"><img src="https://img.shields.io/badge/docs-stable-blue.svg" alt=Stable ></a> <a href="https://BioJulia.github.io/BioServices.jl/dev"><img src="https://img.shields.io/badge/docs-dev-blue.svg" alt=Dev ></a> <a href="https://github.com/BioJulia/BioServices.jl/commits/master/"><img src="https://img.shields.io/github/last-commit/BioJulia/BioServices.jl/master" alt="GitHub last commit (branch)"></a> <a href="https://juliahub.com/ui/Packages/General/BioServices"><img src="https://juliahub.com/docs/General/BioServices/stable/version.svg" alt=version ></a> <a href="https://codecov.io/gh/BioJulia/BioServices.jl"><img src="https://codecov.io/gh/BioJulia/BioServices.jl/branch/master/graph/badge.svg" alt=Coverage ></a> </div> <blockquote> <p>Julia interface to APIs for various bio-related web services </p> </blockquote> <p><a href="https://github.com/BioJulia/BioServices.jl/">BioServices.jl</a></p> <h2 id=ncbiblastjl ><a href="#ncbiblastjl" class=header-anchor >NCBIBlast.jl</a></h2> <div class=badge > <a href="https://github.com/BioJulia/NCBIBlast.jl"><img src="https://img.shields.io/github/stars/BioJulia/NCBIBlast.jl?style=social" alt="GitHub Repo stars"></a> <a href="https://juliahub.com/ui/Packages/General/NCBIBlast?t=2"><img src="https://juliahub.com/docs/General/NCBIBlast/stable/deps.svg" alt=deps ></a> <a href="https://pkgs.genieframework.com?packages=NCBIBlast"><img src="https://shields.io/endpoint?url=https://pkgs.genieframework.com/api/v1/badge/NCBIBlast" alt="NCBIBlast Downloads"></a> <br> <a href="https://github.com/BioJulia/NCBIBlast.jl/commits/master/"><img src="https://img.shields.io/github/last-commit/BioJulia/NCBIBlast.jl/master" alt="GitHub last commit (branch)"></a> <a href="https://juliahub.com/ui/Packages/General/NCBIBlast"><img src="https://juliahub.com/docs/General/NCBIBlast/stable/version.svg" alt=version ></a> <a href="https://codecov.io/gh/BioJulia/NCBIBlast.jl"><img src="https://codecov.io/gh/BioJulia/NCBIBlast.jl/branch/master/graph/badge.svg" alt=Coverage ></a> </div> <blockquote> <p>Thin wrapper around NCBI&#39;s BLAST&#43; CLI https://www.ncbi.nlm.nih.gov/books/NBK569856/</p> </blockquote> <p><a href="https://github.com/BioJulia/NCBIBlast.jl/">NCBIBlast.jl</a></p> <p>From the package README:</p> <p>This package is a thin wrapper around the Basic Local Alignment Search Tool CLI, better known as BLAST, developed by the National Center for Biotechnology Information &#40;NCBI&#41;.</p> <p>For now, this uses CondaPkg.jl to install BLAST&#43;.</p> <h2 id=fastxjl ><a href="#fastxjl" class=header-anchor >FASTX.jl</a></h2> <div class=badge > <a href="https://github.com/BioJulia/FASTX.jl"><img src="https://img.shields.io/github/stars/BioJulia/FASTX.jl?style=social" alt="GitHub Repo stars"></a> <a href="https://juliahub.com/ui/Packages/General/FASTX?t=2"><img src="https://juliahub.com/docs/General/FASTX/stable/deps.svg" alt=deps ></a> <a href="https://pkgs.genieframework.com?packages=FASTX"><img src="https://shields.io/endpoint?url=https://pkgs.genieframework.com/api/v1/badge/FASTX" alt="FASTX Downloads"></a> <br> <a href="https://BioJulia.github.io/FASTX.jl/stable"><img src="https://img.shields.io/badge/docs-stable-blue.svg" alt=Stable ></a> <a href="https://BioJulia.github.io/FASTX.jl/dev"><img src="https://img.shields.io/badge/docs-dev-blue.svg" alt=Dev ></a> <a href="https://github.com/BioJulia/FASTX.jl/commits/master/"><img src="https://img.shields.io/github/last-commit/BioJulia/FASTX.jl/master" alt="GitHub last commit (branch)"></a> <a href="https://juliahub.com/ui/Packages/General/FASTX"><img src="https://juliahub.com/docs/General/FASTX/stable/version.svg" alt=version ></a> <a href="https://codecov.io/gh/BioJulia/FASTX.jl"><img src="https://codecov.io/gh/BioJulia/FASTX.jl/branch/master/graph/badge.svg" alt=Coverage ></a> </div> <blockquote> <p>Parse and process FASTA and FASTQ formatted files of biological sequences.</p> </blockquote> <p><a href="https://github.com/BioJulia/FASTX.jl">FASTX</a></p> <p>FASTX provides I/O and utilities for manipulating FASTA and FASTQ, formatted sequence data files.</p> <h2 id=xamjl ><a href="#xamjl" class=header-anchor >XAM.jl</a></h2> <div class=badge > <a href="https://github.com/BioJulia/XAM.jl"><img src="https://img.shields.io/github/stars/BioJulia/XAM.jl?style=social" alt="GitHub Repo stars"></a> <a href="https://juliahub.com/ui/Packages/General/XAM?t=2"><img src="https://juliahub.com/docs/General/XAM/stable/deps.svg" alt=deps ></a> <a href="https://pkgs.genieframework.com?packages=XAM"><img src="https://shields.io/endpoint?url=https://pkgs.genieframework.com/api/v1/badge/XAM" alt="XAM Downloads"></a> <br> <a href="https://BioJulia.github.io/XAM.jl/stable"><img src="https://img.shields.io/badge/docs-stable-blue.svg" alt=Stable ></a> <a href="https://BioJulia.github.io/XAM.jl/dev"><img src="https://img.shields.io/badge/docs-dev-blue.svg" alt=Dev ></a> <a href="https://github.com/BioJulia/XAM.jl/commits/master/"><img src="https://img.shields.io/github/last-commit/BioJulia/XAM.jl/master" alt="GitHub last commit (branch)"></a> <a href="https://juliahub.com/ui/Packages/General/XAM"><img src="https://juliahub.com/docs/General/XAM/stable/version.svg" alt=version ></a> <a href="https://codecov.io/gh/BioJulia/XAM.jl"><img src="https://codecov.io/gh/BioJulia/XAM.jl/branch/master/graph/badge.svg" alt=Coverage ></a> </div> <blockquote> <p>Parse and process FASTA and FASTQ formatted files of biological sequences. </p> </blockquote> <p><a href="https://github.com/BioJulia/XAM.jl">XAM.jl</a></p> <p>FASTX provides I/O and utilities for manipulating FASTA and FASTQ, formatted sequence data files.</p> <h2 id=pairwisemappingformatjl ><a href="#pairwisemappingformatjl" class=header-anchor >PairwiseMappingFormat.jl</a></h2> <div class=badge > <a href="https://github.com/BioJulia/PairwiseMappingFormat.jl"><img src="https://img.shields.io/github/stars/BioJulia/PairwiseMappingFormat.jl?style=social" alt="GitHub Repo stars"></a> <a href="https://juliahub.com/ui/Packages/General/PairwiseMappingFormat?t=2"><img src="https://juliahub.com/docs/General/PairwiseMappingFormat/stable/deps.svg" alt=deps ></a> <a href="https://pkgs.genieframework.com?packages=PairwiseMappingFormat"><img src="https://shields.io/endpoint?url=https://pkgs.genieframework.com/api/v1/badge/PairwiseMappingFormat" alt="PairwiseMappingFormat Downloads"></a> <br> <a href="https://BioJulia.github.io/PairwiseMappingFormat.jl/stable"><img src="https://img.shields.io/badge/docs-stable-blue.svg" alt=Stable ></a> <a href="https://BioJulia.github.io/PairwiseMappingFormat.jl/dev"><img src="https://img.shields.io/badge/docs-dev-blue.svg" alt=Dev ></a> <a href="https://github.com/BioJulia/PairwiseMappingFormat.jl/commits/master/"><img src="https://img.shields.io/github/last-commit/BioJulia/PairwiseMappingFormat.jl/master" alt="GitHub last commit (branch)"></a> <a href="https://juliahub.com/ui/Packages/General/PairwiseMappingFormat"><img src="https://juliahub.com/docs/General/PairwiseMappingFormat/stable/version.svg" alt=version ></a> <a href="https://codecov.io/gh/BioJulia/PairwiseMappingFormat.jl"><img src="https://codecov.io/gh/BioJulia/PairwiseMappingFormat.jl/branch/master/graph/badge.svg" alt=Coverage ></a> </div> <blockquote> <p>Parser for the PAF format in bioinformatics </p> </blockquote> <p><a href="https://github.com/BioJulia/PairwiseMappingFormat.jl">PairwiseMappingFormat.jl</a></p> <p>PairwiseMappingFormat.jl provide a parser for Pairwise Mapping Format &#40;PAF&#41; files. PAF is a simple, tab-delimited format created by programs such as minimap2.</p> <h2 id=proteinsecondarystructuresjl ><a href="#proteinsecondarystructuresjl" class=header-anchor >ProteinSecondaryStructures.jl</a></h2> <div class=badge > <a href="https://github.com/BioJulia/ProteinSecondaryStructures.jl"><img src="https://img.shields.io/github/stars/BioJulia/ProteinSecondaryStructures.jl?style=social" alt="GitHub Repo stars"></a> <a href="https://juliahub.com/ui/Packages/General/ProteinSecondaryStructures?t=2"><img src="https://juliahub.com/docs/General/ProteinSecondaryStructures/stable/deps.svg" alt=deps ></a> <a href="https://pkgs.genieframework.com?packages=ProteinSecondaryStructures"><img src="https://shields.io/endpoint?url=https://pkgs.genieframework.com/api/v1/badge/ProteinSecondaryStructures" alt="ProteinSecondaryStructures Downloads"></a> <br> <a href="https://BioJulia.github.io/ProteinSecondaryStructures.jl/stable"><img src="https://img.shields.io/badge/docs-stable-blue.svg" alt=Stable ></a> <a href="https://BioJulia.github.io/ProteinSecondaryStructures.jl/dev"><img src="https://img.shields.io/badge/docs-dev-blue.svg" alt=Dev ></a> <a href="https://github.com/BioJulia/ProteinSecondaryStructures.jl/commits/master/"><img src="https://img.shields.io/github/last-commit/BioJulia/ProteinSecondaryStructures.jl/master" alt="GitHub last commit (branch)"></a> <a href="https://juliahub.com/ui/Packages/General/ProteinSecondaryStructures"><img src="https://juliahub.com/docs/General/ProteinSecondaryStructures/stable/version.svg" alt=version ></a> <a href="https://codecov.io/gh/BioJulia/ProteinSecondaryStructures.jl"><img src="https://codecov.io/gh/BioJulia/ProteinSecondaryStructures.jl/branch/master/graph/badge.svg" alt=Coverage ></a> </div> <blockquote> <p>Wrapper to protein secondary structure calculation packages</p> </blockquote> <p><a href="https://github.com/BioJulia/ProteinSecondaryStructures.jl">ProteinSecondaryStructures.jl</a></p> <p>From the package README:</p> <p>This package parses <a href=" http://webclu.bio.wzw.tum.de/stride/">STRIDE</a> and <a href="https://github.com/PDB-REDO/dssp">DSSP</a> secondary structure prediction outputs, to make them convenient to use from Julia, particularly for the analysis of MD simulations. </p> <h2 id=biomakiejl ><a href="#biomakiejl" class=header-anchor >BioMakie.jl</a></h2> <div class=badge > <a href="https://github.com/BioJulia/BioMakie.jl"><img src="https://img.shields.io/github/stars/BioJulia/BioMakie.jl?style=social" alt="GitHub Repo stars"></a> <a href="https://juliahub.com/ui/Packages/General/BioMakie?t=2"><img src="https://juliahub.com/docs/General/BioMakie/stable/deps.svg" alt=deps ></a> <a href="https://pkgs.genieframework.com?packages=BioMakie"><img src="https://shields.io/endpoint?url=https://pkgs.genieframework.com/api/v1/badge/BioMakie" alt="BioMakie Downloads"></a> <br> <a href="https://biojulia.dev/BioMakie.jl/dev/"><img src="https://img.shields.io/badge/docs-dev-blue.svg" alt=Doc ></a> <a href="https://github.com/BioJulia/BioMakie.jl/commits/master/"><img src="https://img.shields.io/github/last-commit/BioJulia/BioMakie.jl/master" alt="GitHub last commit (branch)"></a> <a href="https://juliahub.com/ui/Packages/General/BioMakie"><img src="https://juliahub.com/docs/General/BioMakie/stable/version.svg" alt=version ></a> <a href="https://codecov.io/gh/BioJulia/BioMakie.jl"><img src="https://codecov.io/gh/BioJulia/BioMakie.jl/branch/master/graph/badge.svg" alt=Coverage ></a> </div> <blockquote> <p>Plotting and interface tools for biology. </p> </blockquote> <p><a href="https://github.com/BioJulia/BioMakie.jl">BioMakie.jl</a></p> <p><a href="https://github.com/BioJulia/BioMakie.jl">BioMakie.jl</a> has functions to visualize</p> <ul> <li><p>Protein 3D structures</p> <li><p>Multiple Sequence Alignments</p> </ul> <h2 id=mitosjl ><a href="#mitosjl" class=header-anchor >MIToS.jl</a></h2> <div class=badge > <a href="https://github.com/diegozea/MIToS.jl"><img src="https://img.shields.io/github/stars/diegozea/MIToS.jl?style=social" alt="GitHub Repo stars"></a> <a href="https://juliahub.com/ui/Packages/General/MIToS?t=2"><img src="https://juliahub.com/docs/General/MIToS/stable/deps.svg" alt=deps ></a> <a href="https://pkgs.genieframework.com?packages=MIToS"><img src="https://shields.io/endpoint?url=https://pkgs.genieframework.com/api/v1/badge/MIToS" alt="MIToS Downloads"></a> <br> <a href="https://diegozea.github.io/MIToS.jl/stable"><img src="https://img.shields.io/badge/docs-stable-blue.svg" alt=Stable ></a> <a href="https://diegozea.github.io/MIToS.jl/dev"><img src="https://img.shields.io/badge/docs-dev-blue.svg" alt=Dev ></a> <a href="https://github.com/diegozea/MIToS.jl/commits/master/"><img src="https://img.shields.io/github/last-commit/diegozea/MIToS.jl/master" alt="GitHub last commit (branch)"></a> <a href="https://juliahub.com/ui/Packages/General/MIToS"><img src="https://juliahub.com/docs/General/MIToS/stable/version.svg" alt=version ></a> <a href="https://codecov.io/gh/diegozea/MIToS.jl"><img src="https://codecov.io/gh/diegozea/MIToS.jl/branch/master/graph/badge.svg" alt=Coverage ></a> </div> <blockquote> <p>A Julia package to analyze protein sequences, structures, and evolutionary information </p> </blockquote> <p><a href="https://github.com/diegozea/MIToS.jl">MIToS</a></p> <p>From the package README:</p> <p>MIToS provides a comprehensive suite of tools for the analysis of protein sequences and structures. It allows working with <strong>Multiple Sequence Alignments &#40;MSAs&#41;</strong> to obtain evolutionary information in the Julia language &#91;1&#93;. In particular, it eases the analysis of coevoling position in an MSA using <strong>Mutual Information &#40;MI&#41;</strong>, a measure of covariation. MI-derived scores are good predictors of inter-residue contacts in a protein structure and functional sites in proteins &#91;2,3&#93;. To allow such analysis, MIToS also implements several useful tools for working with protein structures, such as those available in the <strong>Protein Data Bank &#40;PDB&#41;</strong> or predicted by AlphaFold 2.</p> <h2 id=xsimjl ><a href="#xsimjl" class=header-anchor >XSim.jl</a></h2> <div class=badge > <a href="https://github.com/reworkhow/XSim.jl"><img src="https://img.shields.io/github/stars/reworkhow/XSim.jl?style=social" alt="GitHub Repo stars"></a> <a href="https://juliahub.com/ui/Packages/General/XSim?t=2"><img src="https://juliahub.com/docs/General/XSim/stable/deps.svg" alt=deps ></a> <a href="https://pkgs.genieframework.com?packages=XSim"><img src="https://shields.io/endpoint?url=https://pkgs.genieframework.com/api/v1/badge/XSim" alt="XSim Downloads"></a> <br> <a href="https://reworkhow.github.io/XSim.jl/"><img src="https://img.shields.io/badge/docs-XSim.jl-blue.svg" alt=Doc ></a> <a href="https://github.com/reworkhow/XSim.jl/commits/master/"><img src="https://img.shields.io/github/last-commit/reworkhow/XSim.jl/master" alt="GitHub last commit (branch)"></a> <a href="https://juliahub.com/ui/Packages/General/XSim"><img src="https://juliahub.com/docs/General/XSim/stable/version.svg" alt=version ></a> <a href="https://codecov.io/gh/reworkhow/XSim.jl"><img src="https://codecov.io/gh/reworkhow/XSim.jl/branch/master/graph/badge.svg" alt=Coverage ></a> </div> <blockquote> <p>Simulate sequence data and complicated pedigree structures </p> </blockquote> <p><a href="https://github.com/reworkhow/XSim.jl">XSim.jl</a></p> <p>From the package README:</p> <p>XSim is a fast and user-friendly tool to simulate sequence data and complicated pedigree structures.</p> <p>Features</p> <ul> <li><p>An efficient CPOS algorithm</p> <li><p>Using founders that are characterized by real genome sequence data</p> <li><p>Complicated pedigree structures among descendants</p> </ul> <h2 id=genefinderjl ><a href="#genefinderjl" class=header-anchor >GeneFinder.jl</a></h2> <div class=badge > <a href="https://github.com/camilogarciabotero/GeneFinder.jl"><img src="https://img.shields.io/github/stars/camilogarciabotero/GeneFinder.jl?style=social" alt="GitHub Repo stars"></a> <a href="https://juliahub.com/ui/Packages/General/GeneFinder?t=2"><img src="https://juliahub.com/docs/General/GeneFinder/stable/deps.svg" alt=deps ></a> <a href="https://pkgs.genieframework.com?packages=GeneFinder"><img src="https://shields.io/endpoint?url=https://pkgs.genieframework.com/api/v1/badge/GeneFinder" alt="GeneFinder Downloads"></a> <br> <a href="https://camilogarciabotero.github.io/GeneFinder.jl/stable"><img src="https://img.shields.io/badge/docs-stable-blue.svg" alt=Stable ></a> <a href="https://camilogarciabotero.github.io/GeneFinder.jl/dev"><img src="https://img.shields.io/badge/docs-dev-blue.svg" alt=Dev ></a> <a href="https://github.com/camilogarciabotero/GeneFinder.jl/commits/main/"><img src="https://img.shields.io/github/last-commit/camilogarciabotero/GeneFinder.jl/main" alt="GitHub last commit (branch)"></a> <a href="https://juliahub.com/ui/Packages/General/GeneFinder"><img src="https://juliahub.com/docs/General/GeneFinder/stable/version.svg" alt=version ></a> <a href="https://codecov.io/gh/camilogarciabotero/GeneFinder.jl"><img src="https://codecov.io/gh/camilogarciabotero/GeneFinder.jl/branch/main/graph/badge.svg" alt=Coverage ></a> </div> <blockquote> <p>A Gene Finder framework for Julia.</p> </blockquote> <p><a href="https://github.com/camilogarciabotero/GeneFinder.jl">GeneFinder.jl</a> is a species-agnostic, algorithm extensible, sequence-anonymous &#40;genome, metagenomes&#41; <em>gene finder</em> library framework for the Julia Language.</p> <p>From the package README:</p> <p>The <code>GeneFinder</code> package aims to be a versatile module that enables the application of different gene finding algorithms to the <code>BioSequence</code> type, by providing a common interface and a flexible data structure to store the predicted ORFI or genes. The package is designed to be easily extensible, allowing users to implement their own algorithms and integrate them into the framework.</p> <p>This package is currently under development and is not yet ready for production use. The API is subject to change.</p> <h1 id=star_history ><a href="#star_history" class=header-anchor >Star History</a></h1> <div class=star-history > <a href="https://star-history.com/#BioJulia/BioSequences.jl&amp;BioJulia/BioAlignments.jl&amp;BioJulia/BioStructures.jl&amp;BioJulia/GeneticVariation.jl&amp;BioJulia/GenomeGraphs.jl&amp;BioJulia/FASTX.jl&amp;BioJulia/XAM.jl&amp;BioJulia/BioMakie.jl&amp;diegozea/MIToS.jl&amp;Date"><img src="https://api.star-history.com/svg?repos=BioJulia/BioSequences.jl,BioJulia/BioAlignments.jl,BioJulia/BioStructures.jl,BioJulia/GeneticVariation.jl,BioJulia/GenomeGraphs.jl,BioJulia/FASTX.jl,BioJulia/XAM.jl,BioJulia/BioMakie.jl,diegozea/MIToS.jl&amp;type=Date" alt="Star History Chart"> </a> </div> <div class=star-history > <a href="https://star-history.com/#BioJulia/GenomicFeatures.jl&amp;BioJulia/GenomicAnnotations.jl&amp;BioJulia/Phylogenies.jl&amp;BioJulia/NCBIBlast.jl&amp;BioJulia/BioServices.jl&amp;BioJulia/PairwiseMappingFormat.jl&amp;BioJulia/ProteinSecondaryStructures.jl&amp;camilogarciabotero/GeneFinder.jl&amp;reworkhow/XSim.jl&amp;Date"><img src="https://api.star-history.com/svg?repos=BioJulia/GenomicFeatures.jl,BioJulia/GenomicAnnotations.jl,BioJulia/Phylogenies.jl,BioJulia/NCBIBlast.jl,BioJulia/BioServices.jl,BioJulia/PairwiseMappingFormat.jl,BioJulia/ProteinSecondaryStructures.jl,camilogarciabotero/GeneFinder.jl,reworkhow/XSim.jl&amp;type=Date" alt="Star History Chart"> </a> </div> <h1 id=biojl_is_deprecated ><a href="#biojl_is_deprecated" class=header-anchor >Bio.jl is Deprecated</a></h1> <p>Note that the <a href="https://github.com/BioJulia/Bio.jl">Bio.jl</a> package is deprecated. In <a href="https://biojulia.dev/posts/biojl/">this blogpost</a>, the main developer of Bio.jl, describes where the functionalities have gone:</p> <ul> <li><p>Bio.Seq became <a href="https://github.com/BioJulia/BioSequences.jl/">BioSequences.jl</a></p> <li><p>Bio.Align became <a href="https://github.com/BioJulia/BioAlignments.jl/">BioAlignments.jl</a></p> <li><p>Bio.Intervals became <a href="https://github.com/BioJulia/GenomicFeatures.jl/">GenomicFeatures.jl</a></p> <li><p>Bio.Structure became <a href="https://github.com/BioJulia/BioStructures.jl/">BioStructures.jl</a></p> <li><p>Bio.Var became <a href="https://github.com/BioJulia/GeneticVariation.jl/">GeneticVariation.jl</a></p> <li><p>Bio.Phylo became <a href="https://github.com/BioJulia/Phylogenies.jl/">Phylogenies.jl</a></p> <li><p>Bio.Services became <a href="https://github.com/BioJulia/BioServices.jl/">BioServices.jl</a></p> <li><p>Bio.Tools became <a href="https://github.com/BioJulia/BioTools.jl/">BioTools.jl</a> &#40;now archived&#41;</p> </ul> <div class=page-foot > <!-- Original content <a href="http://creativecommons.org/licenses/by-sa/4.0/">CC BY-SA 4.0</a> {{ fill author }}. {{isnotpage /tag/*}}Last modified: {{ fill fd_mtime }}.{{end}} Website built with <a href="https://github.com/tlienart/Franklin.jl">Franklin.jl</a> and the <a href="https://julialang.org">Julia programming language</a>. --> <!-- Modified content--> This website is a community effort covering a lot of ever-changing information. It will therefore never be complete or without error. If you see something wrong, or have something to contribute, please see the "Contributing" section in the <a href="https://github.com/JuliaPackageComparisons/JuliaPackageComparisons.github.io">github repository</a></a>.<br><br> Last modified: August 24, 2025. Built with <a href="https://github.com/tlienart/Franklin.jl">Franklin.jl</a> </div> </div> </div> </div> </div> <!-- end of class page-wrap-->